lib/classes/BaseLanguage.mjs

```diff
@@ -0,0 +1,151 @@
+import AbstractLanguage from './AbstractLanguage.mjs';
+
+/**
+ * This class is used by multiple languages (mostly european).
+ */
+export default class extends AbstractLanguage {
+  /**
+   * Get word for number if it matches a language card.
+   * @param {number} number Card number value.
+   * @returns {string|undefined} Return card word or undefined if no card.
+   */
+  getCardWord(number) {
+    // Get matching card from number
+    const card = this.cards.find(_card => _card[0] == number);
+
+    // Return card word or undefined if no card found
+    return (Array.isArray(card) ? card[1] : undefined);
+  }
+
+  /**
+   * Convert a number value into words sets for further processing.
+   * @param {number} value The number value to convert to cardinal form.
+   * @returns {object} Word sets (and pairs) from value.
+   * @todo Simplify return object.
+   */
+  splitNum(value) {
+    // Loop through all cards
+    // NOTE: Language cards should be ordered from highest to lowest
+    for (let i = 0; i < this.cards.length; i++) {
+      const number = this.cards[i][0];
+
+      // If card number is greater than value skip to next card
+      if (number > value) {
+        continue;
+      }
+
+      const out = [];
+      let div; // Quantity of card set values
+      let mod; // Remaining value
+
+      // Calculate quantity and remaining value
+      // Override variables for 0 as math will fail
+      if (value == 0) {
+        div = 1;
+        mod = 0;
+      } else {
+        div = Math.floor(value / number);
+        mod = value % number;
+      }
+
+      // Is value perfect match of card number?
+      if (div == 1) {
+        // TODO: Merge word set pairs together (if possible) to simplify return object
+        out.push({
+          [this.getCardWord(1)]: 1,
+        });
+      } else {
+        // TODO: Understand the logic for this
```

Comment:
@forzagreen Could you possibly help me understand what's going on here?

Reply:
@TylerVigario I'll try to remember and let you know

Reply:
I didn't find the details of the implementation. But I remember this "cards" part was inspired from [num2words](https://github.com/savoirfairelinux/num2words/blob/master/num2words/base.py)